catalogd/internal/storage/multireadseeker.go

```diff
@@ -0,0 +1,118 @@
+package storage
```

Comment:
Oh, we should discuss this. I added this because the `http` package documentation recommended using `http.ServeContent` over `io.Copy` for a number of reasons, and `http.ServeContent` needs an `io.ReadSeeker`. The main benefit for us would be support for the `Range` http headers.

I found this multiReadSeaker implementation in a Golang issue proposing that it be added to the standard library, so there may be license concerns. We may need to implement our own or look into implementing the range functionality another way.

Reply:
Discussion topic for when we get on a call tomorrow: why deal with all of these ourselves, when others have already done it for us? 

https://github.com/operator-framework/operator-controller/pull/1654

Reply:
Let's just delete this file, stop wrapping the query readers with this, and use `io.Copy` instead of `http.ServeContent`